front/lib/api/assistant/agent_usage.ts

```diff
@@ -0,0 +1,216 @@
+import { AgentUsageType, ModelId } from "@dust-tt/types";
+import { literal, Op } from "sequelize";
+import { v4 as uuidv4 } from "uuid";
+
+import { Mention, Message, UserMessage } from "@app/lib/models";
+import { redisClient } from "@app/lib/redis";
+
+// TTL of the ranking data in Redis.
+const rankingTTL = 60 * 60 * 24 * 30; // 30 days
+
+function _getKeys(agentConfigurationId: string) {
+  // One sorted set per agent for counting the number of times the agent has been used.
+  // score is: timestamp of each use of the agent
+  // value: random unique distinct value.
+  const agentUsedCountKey = `agent_usage_count_${agentConfigurationId}`;
+
+  // One sorted set per agent for counting the number of users that have used the agent.
+  // score is: timestamp of last usage by a given user
+  // value: user_id
+  const agentUserCountKey = `agent_user_count_${agentConfigurationId}`;
+  return {
+    agentUsedCountKey,
+    agentUserCountKey,
+  };
+}
+
+async function signalInRedis({
+  agentConfigurationId,
+  userId,
+  timestamp,
+  redis,
+}: {
+  agentConfigurationId: string;
+  userId: string;
+  timestamp: number;
+  messageId: ModelId;
+  redis: Awaited<ReturnType<typeof redisClient>> | undefined;
+}) {
+  let needRedisQuit = false;
```

Comment:
this variable is not really necesssary is it? it's tied to redis being not null ?

Reply:
Sorry I now see the logic :+1: LGTM